neard/src/config.rs

```diff
@@ -779,12 +783,32 @@ pub fn init_configs(
             config.write_to_file(&dir.join(CONFIG_FILENAME));
 
             // TODO: add download genesis for mainnet
+            #[cfg(feature = "protocol_feature_restore_receipts_after_fix")]
+            let mut genesis: Genesis = serde_json::from_slice(*MAINNET_GENESIS_JSON)
+                .expect("Failed to deserialize MainNet genesis");
+
+            #[cfg(not(feature = "protocol_feature_restore_receipts_after_fix"))]
             let genesis: Genesis = serde_json::from_slice(*MAINNET_GENESIS_JSON)
                 .expect("Failed to deserialize MainNet genesis");
+
             if let Some(account_id) = account_id {
                 generate_validator_key(account_id, &dir.join(config.validator_key_file));
             }
 
+            // Put receipts which were lost because of a bug in apply_chunks to the runtime config.
+            // See https://github.com/near/nearcore/pull/4248/ for more details
+            checked_feature!(
```

Comment:
I think we want to use only bool version of this macro (and if) going forward

Reply:
I wonder if this is correct place to do it at all, this only runs on init I think, not on run, and besides changing genesis is a bad idea

Reply:
Agree, moved it to `MigrationData`